tools/jdbc/src/main/java/org/duckdb/DuckDBPreparedStatement.java

```diff
@@ -164,6 +164,10 @@ public void setObject(int parameterIndex, Object x) throws SQLException {
 		if (params.length == 0) {
 			params = new Object[getParameterMetaData().getParameterCount()];
 		}
+        // Change sql.Timestamp to DuckDBTimestamp
```

Comment:
Does Java's `sql` library have `TIMESTAMPTZ` too?

Reply:
If you know what type of timestamp you will get back, it is possible to use sql.Timestamp in both cases and then make a cast. In my understanding this is legacy since Java 8 and the much better java.time package. Via the get/setObject method an OffsetDateTime can be used for TIMESTAMPZ, but as this is also new in DuckDB I haven't tried it.

I would defer the whole timestamp with timezone to a later PR as it is a new feature.

Reply:
`TIMESTAMPTZ` internally is identical to `TIMESTAMP` so I was thinking just an extra case label was all that was needed. It sounds like Java (v8?) actually treats all timestamps as `TIMESTAMPTZ` and does the part binning in the users' local time zone?

Reply:
Yes, you are right. E.g. a `Timestamp.valueOf("1970-01-01 00:00:00").getTime()` has the result -3600000 in my timezone. This is why it is necessary to normalize the long representation of the TIMESTAMP to UTC inside the database and then back to what ever timezone the client is in.

Implementing the TIMESTAMPTZ should be straight forward, but there is some work left to implement the get/setObject methods.